app/src/main/java/io/appium/uiautomator2/model/UiAutomationElement.java

```diff
@@ -182,4 +182,10 @@ private String getBounds(AccessibilityNodeInfo node) {
     protected Map<Attribute, Object> getAttributes() {
         return attributes;
     }
+
+    @Nullable
+    public String getNullableText() {
+        String text = getText();
+        return (text != null && text.isEmpty()) ? null : text;
```

Comment:
better, but still not ideal. I would prefer to have a special argument for ElementHelper methods, which defines whether null should be replaced or not, so we don't need to perform "double transformation"

Reply:
Feel free to open a PR.

Reply:
sure will do. Thanks for the hint